src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -1761,7 +1797,16 @@ static int do_map(int argc, const char *argv[], Config *cfg, bool reconnect)
     if (r < 0)
       goto close_nbd;
 
-    cout << cfg->devpath << std::endl;
+    std::string cookie;
+    if (use_netlink && cfg->show_cookie && !cfg->cookie.empty()) {
+      cookie = get_cookie(cfg->devpath);
```

Comment:
It would be nice to assert on the returned cookie independent of `--show-cookie`.  Could you restructure these two `if` statements so that `get_cookie()` is called and verified on `use_netlink && !cfg->cookie.empty()`?